## How to split by character - Character Text Splitter 

This is the simplest method. This splits based on a given character sequences, which defaults to "\n\n". Chunk length is measured by number of characters.

1. How the text is split: by single character seperator.
2. How the chunk size is measured: by number of characters.

In [1]:
from langchain_community.document_loaders import TextLoader

loader = TextLoader('speech.txt')
docs = loader.load()
docs

[Document(metadata={'source': 'speech.txt'}, page_content='Dr. **Abdul Kalam**, the 11th President of India and a renowned scientist, delivered many inspiring speeches during his lifetime. One of his most famous speeches was delivered at **IIT Kharagpur** in 2002, titled "You Have to Dream Before Your Dreams Can Come True." This speech is filled with wisdom, encouragement, and motivational messages for the youth, urging them to aspire for greatness and contribute to society. Here’s an excerpt from one of his most inspiring speeches:\n\n---\n\n### **Excerpt from Dr. Abdul Kalam\'s Speech at IIT Kharagpur (2002)**\n\n**"Dream, Dream, Dream!"**\n\n_"Dream, Dream, Dream. Dreams transform into thoughts, and thoughts result in action."_\n\n**Address to Students at IIT Kharagpur, 2002**\n\nDr. Kalam began his speech by talking about the importance of dreaming big and striving to achieve those dreams. He emphasized that **dreams** are the foundation of any success, and they are essential for p

In [3]:
from langchain_text_splitters import CharacterTextSplitter

text_splitter = CharacterTextSplitter(separator="\n\n", chunk_size=100, chunk_overlap=20)
text_splitter.split_documents(docs)

Created a chunk of size 479, which is longer than the specified 100
Created a chunk of size 243, which is longer than the specified 100
Created a chunk of size 359, which is longer than the specified 100
Created a chunk of size 116, which is longer than the specified 100
Created a chunk of size 206, which is longer than the specified 100
Created a chunk of size 244, which is longer than the specified 100
Created a chunk of size 200, which is longer than the specified 100
Created a chunk of size 297, which is longer than the specified 100
Created a chunk of size 153, which is longer than the specified 100
Created a chunk of size 351, which is longer than the specified 100
Created a chunk of size 353, which is longer than the specified 100
Created a chunk of size 181, which is longer than the specified 100
Created a chunk of size 216, which is longer than the specified 100
Created a chunk of size 156, which is longer than the specified 100
Created a chunk of size 256, which is longer tha

[Document(metadata={'source': 'speech.txt'}, page_content='Dr. **Abdul Kalam**, the 11th President of India and a renowned scientist, delivered many inspiring speeches during his lifetime. One of his most famous speeches was delivered at **IIT Kharagpur** in 2002, titled "You Have to Dream Before Your Dreams Can Come True." This speech is filled with wisdom, encouragement, and motivational messages for the youth, urging them to aspire for greatness and contribute to society. Here’s an excerpt from one of his most inspiring speeches:'),
 Document(metadata={'source': 'speech.txt'}, page_content="---\n\n### **Excerpt from Dr. Abdul Kalam's Speech at IIT Kharagpur (2002)**"),
 Document(metadata={'source': 'speech.txt'}, page_content='**"Dream, Dream, Dream!"**'),
 Document(metadata={'source': 'speech.txt'}, page_content='_"Dream, Dream, Dream. Dreams transform into thoughts, and thoughts result in action."_'),
 Document(metadata={'source': 'speech.txt'}, page_content='**Address to Students

`Created a chunk of size 479, which is longer than the specified 100`

What is the reason it has created the chunk size of 479 which is longer than 100 ?

- because it did not found the separator, simple

In [7]:
speech = ""
with open("speech.txt") as f:
    speech = f.read()

text_splitter = CharacterTextSplitter(chunk_size=100, chunk_overlap=20)
text = text_splitter.create_documents([speech])

print(text[0])
print(text[1])

Created a chunk of size 479, which is longer than the specified 100
Created a chunk of size 243, which is longer than the specified 100
Created a chunk of size 359, which is longer than the specified 100
Created a chunk of size 116, which is longer than the specified 100
Created a chunk of size 206, which is longer than the specified 100
Created a chunk of size 244, which is longer than the specified 100
Created a chunk of size 200, which is longer than the specified 100
Created a chunk of size 297, which is longer than the specified 100
Created a chunk of size 153, which is longer than the specified 100
Created a chunk of size 351, which is longer than the specified 100
Created a chunk of size 353, which is longer than the specified 100
Created a chunk of size 181, which is longer than the specified 100
Created a chunk of size 216, which is longer than the specified 100
Created a chunk of size 156, which is longer than the specified 100
Created a chunk of size 256, which is longer tha

page_content='Dr. **Abdul Kalam**, the 11th President of India and a renowned scientist, delivered many inspiring speeches during his lifetime. One of his most famous speeches was delivered at **IIT Kharagpur** in 2002, titled "You Have to Dream Before Your Dreams Can Come True." This speech is filled with wisdom, encouragement, and motivational messages for the youth, urging them to aspire for greatness and contribute to society. Here’s an excerpt from one of his most inspiring speeches:'
page_content='---

### **Excerpt from Dr. Abdul Kalam's Speech at IIT Kharagpur (2002)**'
